.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,248 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+name: integration-tests
+
+on:
+  # Run every day at 03:00
+  schedule:
+    - cron: "0 3 * * *"
+  push:
+    paths:
+      - '.github/workflows/integration-tests.yml'
+      - '**/test/**/*IT.java'
+      - '**/test/**/*ITCase.java'
+      - '**/test/**/IT*.java'
+  pull_request:
+    paths:
+      - '.github/workflows/integration-tests.yml'
+      - '**/test/**/*IT.java'
+      - '**/test/**/*ITCase.java'
+      - '**/test/**/IT*.java'
+
+env:
+  DEFAULT_MAVEN_OPTS: >-
+    -Xmx4g
+    -XX:ReservedCodeCacheSize=1g
+    -XX:+UseG1GC
+    -Dorg.slf4j.simpleLogger.defaultLogLevel=WARN
+    -Daether.connector.http.retryHandler.count=5
+    -Daether.connector.http.connectionMaxTtl=30
+    -Duser.language=en
+    -Duser.country=GB
+    -Duser.timezone=UTC
+  MAVEN_COMMAND: >-
+    ./mvnw
+    -V
+    -nsu
+    -ntp
+    -ff
+
+# skip assemblies and modules covered separately by the system-tests workflow
+# skip unit tests (surefire), run integration tests (failsafe), python ITs included as part of the system-tests workflow
+  MAVEN_BUILD_ARGUMENTS: >-
+    -D failsafe.failIfNoSpecifiedTests=false
+    verify
+    -P skip-unit-tests
+    -P integration-tests
+    -P nifi-registry-integration-tests
+  MAVEN_BUILD_EXCLUDE_PROJECTS: >-
+    -pl -:minifi-assembly
+    -pl -:minifi-c2-assembly
+    -pl -:nifi-assembly
+    -pl -:nifi-toolkit-assembly
+    -pl -:nifi-registry-assembly
+    -pl -:nifi-registry-toolkit-assembly
+    -pl -:nifi-runtime-manifest
+    -pl -:nifi-runtime-manifest-test
+    -pl -:nifi-stateless-assembly
+    -pl -:nifi-stateless-processor-tests
+    -pl -:nifi-stateless-system-test-suite
+    -pl -:nifi-system-test-suite
+    -pl -:nifi-nar-provider-assembly
+    -pl -:nifi-py4j-integration-tests
+# skip individual failing ITs (some won't work in a CI environment)
+# always run the (failsafe default) IT*, *IT, *ITCase integration-tests
+  MAVEN_FAILSAFE_SKIP_ITS: >-
```

Comment:
Did you consider enumerating these failing tests in a profile, or is that not possible at a root level? It would be helpful to run a similar profile in a local build environment.

Reply:
I agree that it would be better to make it easier to run the same set of tests locally, so I'll try adding a new profile, e.g. `integration-tests-ci`

Reply:
Thanks @ChrisSamo632!

Reply:
Actually, the [test](https://maven.apache.org/surefire/maven-failsafe-plugin/integration-test-mojo.html#test) configuration allows us to do this - I'd have preferred the use of `includes` and `excludes` I think, but with `test` we at least retain the method-level filtering